code/src/components/TodoInput.js

```diff
@@ -0,0 +1,48 @@
+import React, { useEffect, useState } from "react";
+import { useDispatch } from "react-redux";
+import { todos } from "../reducers/todos.js";
+
+
+export const TodoInput = ({ listId }) => {
+  const dispatch = useDispatch();
+  const [inputValue, setInputValue] = useState("");
+
+  const handleOnSubmit = e => {
+    // Prevent page reload
```

Comment:
right now you can save an empty todo. to prevent this, check if the input is empty and if it is, don't dispatch the addTodo action. 
`if (!inputValue) return`
Another solution is to disable the add button until the input is filled in.